compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1295,8 +1295,15 @@ private STNode parseOptionalRelativePath(boolean isObjectMember) {
         STNode resourcePath;
         STToken nextToken = peek();
         switch (nextToken.kind) {
-            case DOT_TOKEN:
             case IDENTIFIER_TOKEN:
+                if (nextToken.isMissing() && getNextNextToken().kind == SyntaxKind.SLASH_TOKEN) {
+                    // special case `/` to improve the error message for `/hello`
+                    consume(); // consume previously recovered identifier token
+                    addInvalidTokenToNextToken(consume());
```

Comment:
With special-cased, new diagnostic going to be `Invalid token '/'`. Since we are special casing it anyway and this is a common user mistake, we can give a dedicated error msg also. Something like `relative resource path should not begin with '/'`.  

https://github.com/ballerina-platform/ballerina-lang/blob/04f2c2ea37880970f26bb33ce15c10d7af1f4a8e/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L6880
Here we check if `(token.isMissing() && getNextNextToken().kind == SyntaxKind.SLASH_TOKEN)`.
```java
consume(); // to ignore current missing identifier diagnostic
return SyntaxErrors
      .createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,DiagnosticErrorCode.NEW_ERROR);
```
WDYT?

Reply:
+1 will update